Teensy/src/sdCard/sdCard.cpp

```diff
@@ -5,11 +5,135 @@
  * Detailed description, etc.
  */
 
+
 #include <Arduino.h>
+#include <SD.h>
+#include <SPI.h>
+#include <TimeLib.h>
+#include <DS1307RTC.h>
+
 #include "sdCard.h"
 
+File dataFile;
+String timeStamp;
+char * fileName;
+const int chipSelect = BUILTIN_SDCARD;
+
+
 extern bool sendDataToSdCard(sensorData_s *pSensorData)
 {
-    // Send data to SD card
+
+fileName = (char*) timeStamp.c_str();
+dataFile = SD.open(fileName, FILE_WRITE);
+
+  //if there was no problem opening the file => write to it
+  if (dataFile) {
+    Serial.print("writing to");
+    Serial.println(fileName);
+    
+    //Print the data to the file
+    dataFile.print(String(pSensorData->wheelSpeed_mph.fr) + ",");
+    dataFile.print(String(pSensorData->wheelSpeed_mph.fl) + ",");
+    dataFile.print(String(pSensorData->wheelSpeed_mph.rr) + ",");
+    dataFile.print(String(pSensorData->wheelSpeed_mph.rl) + ",");
+
+    dataFile.print(String(pSensorData->engineRev_rpm) + ",");
+
+    dataFile.print(String(pSensorData->damperPos_mm.fr) + ",");
+    dataFile.print(String(pSensorData->damperPos_mm.fl) + ",");
+    dataFile.print(String(pSensorData->damperPos_mm.rr) + ",");
+    dataFile.print(String(pSensorData->damperPos_mm.rl) + ",");
+
+    dataFile.print(String(pSensorData->gearPos) + ",");
+
+    dataFile.print(String(pSensorData->steeringWheelPos_degrees) + ",");
+
+    dataFile.print(String(pSensorData->gyro.x) + ",");
+    dataFile.print(String(pSensorData->gyro.y) + ",");
+    dataFile.print(String(pSensorData->gyro.z) + ",");
+
+    dataFile.print(String(pSensorData->batteryVoltage_dV) + ",");
+
+    dataFile.print(String(pSensorData->throttlePos_mm) + ",");
+
+    dataFile.print(String(pSensorData->fuelPressure_pa) + ",");
+
+    dataFile.println(String(pSensorData->time_ms) + ",");
+
+    dataFile.close();
+  } else {
+    //if the file couldn't be opend => print error message
+    Serial.print("error oppening the file :");
```

Comment:
Move error part to the start of the if statement, I think its a little bit neater to handle the errors first :)